Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -779,13 +779,39 @@ public struct RenderNodeTranslator: SemanticVisitor {
             
             return seeAlsoSections
         } ?? .init(defaultValue: [])
+
+        if let callToAction = article.metadata?.callToAction {
+            if let url = callToAction.url {
+                let downloadIdentifier = RenderReferenceIdentifier(url.description)
+                node.sampleDownload = .init(
+                    action: .reference(
+                        identifier: downloadIdentifier,
+                        isActive: true,
+                        overridingTitle: callToAction.buttonLabel(),
+                        overridingTitleInlineContent: nil))
+                downloadReferences[url.description] = DownloadReference(
+                    identifier: downloadIdentifier,
+                    renderURL: url,
+                    sha512Checksum: ""
+                )
+            } else if let fileReference = callToAction.file {
+                let downloadIdentifier = createAndRegisterRenderReference(forMedia: fileReference, assetContext: .download)
+                node.sampleDownload = .init(action: .reference(
+                    identifier: downloadIdentifier,
+                    isActive: true,
+                    overridingTitle: callToAction.buttonLabel(),
+                    overridingTitleInlineContent: nil
+                ))
+            }
+        }
         
         collectedTopicReferences.append(contentsOf: contentCompiler.collectedTopicReferences)
         node.references = createTopicRenderReferences()
 
         addReferences(imageReferences, to: &node)
         addReferences(videoReferences, to: &node)
         addReferences(linkReferences, to: &node)
+        addReferences(downloadReferences, to: &node)
```

Comment:
I guess we need this because we've never had download references in article pages before?

Reply:
Correct; currently only tutorials use the `downloadReferences` collection.